src/trexio.c

```diff
@@ -213,14 +216,17 @@ trexio_exit_code trexio_write_nucleus_coord(trexio_t* file, const double* coord)
   int64_t dim_coord = nucleus_num*3;
   if (dim_coord < 0) return TREXIO_FAILURE;
 
+  int rank = 2;
```

Comment:
Can you use uint32_t for the rank instead of int?

Reply:
Done: here and below

Reply:
:+1: